assets/js/workflow-reducer.test.js

```diff
@@ -27,62 +45,216 @@ describe('Reducer actions', () => {
   };
 
   // Stub result to be returned by our stub loadWorkflow
-  const mock_load_workflow_result = { id: 1001 };
+  //const mock_load_workflow_result = { id: 1001 };
+  const mock_add_wf_module_result = { id: 40, insert_before: 2 };
 
-  // Mocks state and api globals in for this module
-  beforeEach(() => {
-    store = {
-      getState: () => test_state,
-      dispatch: jest.fn()
+  it('Returns the state if we feed garbage to the reducer', () => {
+    let state = workflowReducer(test_state, {
+      type: 'An ill-advised request',
+      payload: {
+        blob: 'malware.exe'
+      }
+    });
+    expect(state).toBe(test_state);
+  });
+
+  // ADD_MODULE
+  it('Adds a module', () => {
+    let state = workflowReducer(test_state, {
+      type: 'ADD_MODULE_FULFILLED',
+      payload: mock_add_wf_module_result,
+    });
+    expect(state.workflow.wf_modules[2].id).toEqual(40);
+    expect(state.selected_wf_module).toEqual(40);
+  });
+
+  it('Deletes a module', () => {
```

Comment:
Would be good to have tests for boundary conditions, so that all code paths are exercised, e.g. delete last remaining module, delete currently selected module